python-packages/order_utils/src/zero_ex/order_utils/__init__.py

```diff
@@ -159,7 +187,202 @@ def int_to_32_big_endian_bytes(i: int):
+            "LENGTH_0_REQUIRED",
+            "LENGTH_65_REQUIRED",
+        ]
+        for known_revert_reason in known_revert_reasons:
+            if known_revert_reason in str(exception):
+                return (False, known_revert_reason)
+        return (False, f"Unknown: {exception}")
+
+
+class ECSignature(TypedDict):
+    """Object representation of an elliptic curve signature's parameters."""
+
+    v: int
+    r: str
+    s: str
+
+
+def _parse_signature_hex_as_vrs(signature_hex: str) -> ECSignature:
+    """Parse signature hex as a concatentation of EC parameters ordered V, R, S.
+
+    >>> _parse_signature_hex_as_vrs('0x1b117902c86dfb95fe0d1badd983ee166ad259b27acb220174cbb4460d872871137feabdfe76e05924b484789f79af4ee7fa29ec006cedce1bbf369320d034e10b03')
+    {'v': 27, 'r': '117902c86dfb95fe0d1badd983ee166ad259b27acb220174cbb4460d87287113', 's': '7feabdfe76e05924b484789f79af4ee7fa29ec006cedce1bbf369320d034e10b'}
+    """  # noqa: E501 (line too long)
+    signature: ECSignature = {
+        "v": int(signature_hex[2:4], 16),
+        "r": signature_hex[4:68],
+        "s": signature_hex[68:132],
+    }
+    if signature["v"] == 0 or signature["v"] == 1:
+        signature["v"] = signature["v"] + 27
+    return signature
+
+
+def _parse_signature_hex_as_rsv(signature_hex: str) -> ECSignature:
+    """Parse signature hex as a concatentation of EC parameters ordered R, S, V.
+
+    >>> _parse_signature_hex_as_rsv('0x117902c86dfb95fe0d1badd983ee166ad259b27acb220174cbb4460d872871137feabdfe76e05924b484789f79af4ee7fa29ec006cedce1bbf369320d034e10b00')
+    {'r': '117902c86dfb95fe0d1badd983ee166ad259b27acb220174cbb4460d87287113', 's': '7feabdfe76e05924b484789f79af4ee7fa29ec006cedce1bbf369320d034e10b', 'v': 27}
+    """  # noqa: E501 (line too long)
+    signature: ECSignature = {
+        "r": signature_hex[2:66],
+        "s": signature_hex[66:130],
+        "v": int(signature_hex[130:132], 16),
+    }
+    if signature["v"] == 0 or signature["v"] == 1:
+        signature["v"] = signature["v"] + 27
+    return signature
+
+
+def _convert_ec_signature_to_vrs_hex(signature: ECSignature) -> str:
+    """Convert elliptic curve signature object to hex hash string.
+
+    >>> _convert_ec_signature_to_vrs_hex(
+    ...     {
+    ...         'r': '117902c86dfb95fe0d1badd983ee166ad259b27acb220174cbb4460d87287113',
+    ...         's': '7feabdfe76e05924b484789f79af4ee7fa29ec006cedce1bbf369320d034e10b',
+    ...         'v': 27
+    ...     }
+    ... )
+    '0x1b117902c86dfb95fe0d1badd983ee166ad259b27acb220174cbb4460d872871137feabdfe76e05924b484789f79af4ee7fa29ec006cedce1bbf369320d034e10b'
+    """  # noqa: E501 (line too long)
+    return (
+        "0x"
+        + signature["v"].to_bytes(1, byteorder="big").hex()
+        + signature["r"]
+        + signature["s"]
+    )
+
+
+def sign_hash(
+    provider: BaseProvider, signer_address: str, order_hash_hex: str
+) -> str:
+    """Sign a message with the order hash, and return the signature.
```

Comment:
Update so it doesn't mention `orderHash` definitively. It should be an example of the type of hash one could sign.